File: C++/471.cpp
#include <iostream>

using namespace std;

// 471 - Buscando el Pinchazo - Golgotha's Melancholy - https://github.com/Golgotha-sMelancholy

int main()
{
    int loop = 0;
    cin >> loop;
    for (int i = 0; i<loop; i++) {
        int positionA = 0;
        int positionB = 0;
        cin >> positionA;
        cin >> positionB;
        if ((positionA - positionB - 360) % 360 < (positionB - positionA - 360) % 360) {
                cout << "DESCENDENTE" << endl;
            } else if ((positionA - positionB - 360) % 360 > (positionB - positionA - 360) % 360) {
                cout << "ASCENDENTE" << endl;
            } else {
                cout << "DA IGUAL" << endl;
            }
        }
}

File: Java/471.java
import java.util.Scanner;

/**
 *
 * 471 - Buscando el Pinchazo - Iván - https://github.com/wildfireOfMine
 */
public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int prueba = sc.nextInt();
        for (int i = 1; i <= prueba; i++) {
            int posicion = sc.nextInt();
            int pinchazo = sc.nextInt();

            int distancia = (posicion - pinchazo - 360) % 360;
            int equidad = (pinchazo - posicion - 360) % 360;

            if (distancia < equidad) {
                System.out.println("DESCENDENTE");
            } else if (distancia > equidad) {
                System.out.println("ASCENDENTE");
            } else {
                System.out.println("DA IGUAL");
            }

        }
    }
}
